doc/maximal.xml

```diff
@@ -10,102 +11,194 @@
+        subsemigroups as well.  Using <C>MaximalSubsemigroups</C> with the
+        option <C><A>opts</A>.number</C> set to be <K>true</K>, whilst
+        specifying additional options, allows for the extension of the
+        functionality provided by <Ref Attr="NumberMaximalSubsemigroups"/>.
+
+        <P/>
+
+        If <C><A>opts</A>.number</C> is <K>false</K>, then (as per
+        <C><A>opts</A>.gens</C>) <C>MaximalSubsemigroups</C> will return either
+        the maximal subsemigroups of <A>S</A> themselves, or a generating set
+        for each maximal subsemigroup of <A>S</A>.
+
+        <P/>
+
+        The default value for this option is <K>false</K>.
+      </Item>
+
+      <Mark>gens</Mark>
+      <Item>
+        <E>This option is only relevant if <C><A>opts</A>.number</C> is
+          <K>false</K> (or unspecified).</E>
+
+        <P/>
+
+        If <C><A>opts</A>.gens</C> is <K>true</K>, then
+        <C>MaximalSubsemigroups</C> will return a generating set for each
+        maximal subsemigroup that it finds.  If <C><A>opts</A>.gens</C> is
+        <K>false</K>, then the maximal subsemigroups will instead be returned as
+        semigroup objects.
+
+        <P/>
+
+        For certain types of semigroup and maximal subsemigroup, it can be more
+        computationally expensive to find a generating set for the maximal
+        subsemigroup than to create the maximal subsemigroup as a semigroup
+        object without knowing a generating set. The default value for this
+        option is <K>false</K>.
+      </Item>
+
+      <Mark>contain</Mark>
+      <Item>
+        If <C><A>opts</A>.contain</C> is a subset of <A>S</A>, then
+        <C>MaximalSubsemigroups</C> will search for those maximal subsemigroups
+        of <A>S</A> which contain the set <C><A>opts</A>.contain</C>.
+      </Item>
+
+      <Mark>D</Mark>
+      <Item>
+        For a maximal subsemigroup <M>M</M> of a finite semigroup <A>S</A>,
+        there exists a unique &D;-class which contains <M><A>S</A> \setminus
+          M</M>. In other words, the elements of <A>S</A> which <M>M</M> lacks
+        are contained in a unique &D;-class.
+
+        <P/>
+
+        If <C><A>opts</A>.D</C> is a &D;-class of <A>S</A>, then
+        <C>MaximalSubsemigroups</C> will search exclusively for those maximal
+        subsemigroups of <A>S</A> whose complement is contained in
+        <C><A>opts</A>.D</C>.
+      </Item>
+
+      <Mark>types</Mark>
+      <Item>
+        <E>This option is only relevant if <A>S</A> is a Rees (0-)matrix
+          semigroup.</E>
+
+        <P/>
+
+        A maximal subsemigroup of a regular Rees 0-matrix semigroup over a group
+        is one of 6 possible types: 1, it consists of the 0; 2, it consists of
+        everything except the zero; 3, it is a union of rows; 4, it is a union
+        of columns; 5, it is a union of both rows and columns; 6, it intersects
+        every H-class non-trivially. A maximal subsemigroup of a Rees matrix
```

Comment:
`H-class` should be `&H;-class`, please check there aren't more occurrences!

Also: in this view I can't see if you have defined the types anywhere prior to this. I guess you have, but please double check.